river/command/input.zig

```diff
@@ -0,0 +1,121 @@
+            util.gpa.destroy(node);
+        }
+    }
+
+    if (mem.eql(u8, "events", args[2])) {
+        input_config.event_state = try parse(InputConfig.EventState, args[3]);
+    } else if (mem.eql(u8, "accel-profile", args[2])) {
+        input_config.accel_profile = try parse(InputConfig.AccelProfile, args[3]);
+    } else if (mem.eql(u8, "click-method", args[2])) {
+        input_config.click_method = try parse(InputConfig.ClickMethod, args[3]);
+    } else if (mem.eql(u8, "drag", args[2])) {
+        input_config.drag_state = try parse(InputConfig.DragState, args[3]);
+    } else if (mem.eql(u8, "drag-lock", args[2])) {
+        input_config.drag_lock = try parse(InputConfig.DragLock, args[3]);
+    } else if (mem.eql(u8, "disable-while-typing", args[2])) {
+        input_config.dwt_state = try parse(InputConfig.DwtState, args[3]);
+    } else if (mem.eql(u8, "middle-emulation", args[2])) {
+        input_config.middle_emulation = try parse(InputConfig.MiddleEmulation, args[3]);
+    } else if (mem.eql(u8, "natural-scroll", args[2])) {
+        input_config.natural_scroll = try parse(InputConfig.NaturalScroll, args[3]);
+    } else if (mem.eql(u8, "left-handed", args[2])) {
+        input_config.left_handed = try parse(InputConfig.LeftHanded, args[3]);
+    } else if (mem.eql(u8, "tap", args[2])) {
+        input_config.tap_state = try parse(InputConfig.TapState, args[3]);
+    } else if (mem.eql(u8, "tap-button-map", args[2])) {
+        input_config.tap_button_map = try parse(InputConfig.TapButtonMap, args[3]);
+    } else if (mem.eql(u8, "pointer-accel", args[2])) {
+        input_config.pointer_accel = InputConfig.PointerAccel{
+            .value = std.math.clamp(
+                try std.fmt.parseFloat(f32, args[3]),
+                @as(f32, -1.0),
+                @as(f32, 1.0),
+            ),
+        };
+    } else if (mem.eql(u8, "scroll-method", args[2])) {
+        input_config.scroll_method = try parse(InputConfig.ScrollMethod, args[3]);
+    } else if (mem.eql(u8, "scroll-button", args[2])) {
+        const event_code_name = try std.cstr.addNullByte(allocator, args[3]);
+        defer allocator.free(event_code_name);
+        const ret = c.libevdev_event_code_from_name(c.EV_KEY, event_code_name);
+        if (ret < 1) return Error.InvalidButton;
+        input_config.scroll_button = InputConfig.ScrollButton{ .button = @intCast(u32, ret) };
+    }
+
+    // Update matching existing input devices.
+    var it = server.input_manager.input_devices.first;
+    while (it) |device_node| : (it = device_node.next) {
+        if (mem.eql(u8, device_node.data.identifier, args[1])) {
+            input_config.apply(&device_node.data);
+            // We don't break here because it is common to have multiple input
+            // devices with the same identifier.
+        }
+    }
+}
+
+fn parse(comptime T: type, str: []const u8) !T {
+    if (std.meta.stringToEnum(T, str)) |ret| {
+        if (ret != .unset) return ret;
```

Comment:
We could avoid this check (and the existence of this wrapper function) by removing the `unset` field from the enums and making the fields of `InputConfig` `?EnumFoo` instead, defaulting to `null`.